onnxruntime/contrib_ops/rocm/bert/gemm_fast_gelu.cc

```diff
@@ -48,34 +48,23 @@ Status GemmFastGelu<T>::ComputeInternal(OpKernelContext* ctx) const {
   MatMulComputeHelper helper;
   ORT_RETURN_IF_ERROR(helper.Compute(X->Shape(), W->Shape(), transa, transb, trans_batch_a, trans_batch_b, false));
 
-  auto gemm_buffer = GetScratchBuffer<T>(helper.OutputShape().Size());
   Tensor* Y = ctx->Output(0, helper.OutputShape());
 
   // Bail out early if the output is going to be empty
   if (Y->Shape().Size() == 0)
     return Status::OK();
 
-  const float alpha = 1.0f;
-  const float zero = 0.0f;
-
-  if (MatMulImpl<T>(this, helper, reinterpret_cast<const T*>(X->Data<T>()),
-                    reinterpret_cast<const T*>(W->Data<T>()),
-                    reinterpret_cast<T*>(gemm_buffer.get()),
-                    X->Shape(), W->Shape(),
-                    transa, transb, trans_batch_a, trans_batch_b, alpha, zero) != Status::OK()) {
-    return Status(common::ONNXRUNTIME, common::FAIL);
-  }
-
-  int64_t fast_gelu_input_length = Y->Shape().Size();
-  int64_t bias_length = (nullptr == bias) ? 0 : bias->Shape().Size();
-
-  return LaunchFastGeluKernel<HipT>(Stream(),
-                                  static_cast<int>(fast_gelu_input_length),
-                                  static_cast<int>(bias_length),
-                                  reinterpret_cast<HipT*>(gemm_buffer.get()),
-                                  (nullptr != bias) ? reinterpret_cast<const HipT*>(bias->Data<T>()) : nullptr,
-                                  reinterpret_cast<HipT*>(Y->MutableData<T>()),
-                                  false);
+  return LaunchGemmFastGeluKernel<HipT>(
+      IsTunableOpEnabled(),
+      Stream(), RocblasHandle(),
+      transa, transb,
+      static_cast<int64_t>(helper.M()), static_cast<int64_t>(helper.N()), static_cast<int64_t>(helper.K()),
+      alpha,
+      reinterpret_cast<const HipT*>(X->Data<T>()), static_cast<int64_t>(helper.Lda(transa)),
+      reinterpret_cast<const HipT*>(W->Data<T>()), static_cast<int64_t>(helper.Ldb(transb)),
```

Comment:
Why do you `reinterpret_cast<const HipT*>(X->Data<T>())` instead of just `X->Data<HipT>()`